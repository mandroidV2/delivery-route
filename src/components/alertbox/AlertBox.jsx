import React from "react";
import './AlertBox.css'

const AlertBox = (props) => {
    return (
        <div id="myModal" className="modal" 
            style={{display : props.show ? 'block' : 'none'}}>
            <div className="modal-content">
                <p>{props.alert}</p>
                <button onClick={() => props.onHideAlert()}>OK</button>
            </div>
        </div>
    );
};
export default AlertBox;