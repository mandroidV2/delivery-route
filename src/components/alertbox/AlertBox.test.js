import React from "react";
import { shallow } from "enzyme";
import AlertBox from "./AlertBox";

/** Test case is used to check the alert box component  */
describe("AlertBox", () => {
  const component = shallow(<AlertBox />);
  it("should be defined", () => {
    expect(component).toBeDefined();
  });
  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
