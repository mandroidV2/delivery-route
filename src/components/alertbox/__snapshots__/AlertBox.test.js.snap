// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal-content"
      >
        <p />
        <button
          onClick={[Function]}
        >
          OK
        </button>
      </div>,
      "className": "modal",
      "id": "myModal",
      "style": Object {
        "display": "none",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p />,
          <button
            onClick={[Function]}
          >
            OK
          </button>,
        ],
        "className": "modal-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OK",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "OK",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal-content"
        >
          <p />
          <button
            onClick={[Function]}
          >
            OK
          </button>
        </div>,
        "className": "modal",
        "id": "myModal",
        "style": Object {
          "display": "none",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p />,
            <button
              onClick={[Function]}
            >
              OK
            </button>,
          ],
          "className": "modal-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OK",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "OK",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
